pkg/actor/validate_version.go

```diff
@@ -150,27 +150,42 @@ func (v *versionChecker) Act(ctx context.Context, cluster *resource.Cluster) err
 			return err
 		}
 	}
+
+	// We have hit an edge case were sometimes the job selector is nil, the following block is extra code
+	// that tries to list the jobs and then get the job again, which probably will reconcile
+	// the API.  We also removed setting the job selector as well.
 	if job == nil || job.Spec.Selector == nil {
```

Comment:
`job` can never be `nil` we assign it on line line 139, or am I missing something?

Reply:
The “Get” could assign it to nil

Reply:
How does that work? In the `Get` we're passing in a pointer so that the `Get` can modify fields inside of `job` but inside of `Get` I can't assign `obj = nil` and then have `job` in calling scope become `nil`, what am I missing? Is `obj` in `Get` actually a `**Job`? In the comments it just says pointer `obj must be a struct pointer`

Reply:
Changed :)